src/hooks/useGetQueryParamOrReduxValue.ts

```diff
@@ -201,22 +199,30 @@ const useGetQueryParamOrReduxValue = (
   }
   // @ts-ignore
   const reduxValue = useSelector(...useSelectorArguments);
-  // TODO: this bit is identical to the one in useGetQueryParamOrXstateValue.ts, keep it DRY
+  const surahReduxValue = useSelector(
+    QUERY_PARAMS_DATA[QueryParam.SURAH].reduxSelector,
+    QUERY_PARAMS_DATA[QueryParam.SURAH].reduxEqualityFunction,
+  );
   // if the param exists in the url
   if (isReady && query[queryParam] !== undefined) {
-    const { validate, valueType } = QUERY_PARAMS_DATA[queryParam];
-
+    const { valueType, validate } = QUERY_PARAMS_DATA[queryParam];
     const paramStringValue = String(query[queryParam]);
     const isValidValue = validate(paramStringValue, chaptersData, query);
+    const parsedQueryParamValue = getQueryParamValueByType(paramStringValue, valueType);
+    const isQueryParamDifferent = getIsQueryParamDifferent(paramStringValue, valueType, reduxValue);
+
+    if (queryParam === QueryParam.VERSE_TO || queryParam === QueryParam.VERSE_FROM) {
+      return {
+        value: getVerseValue(query, queryParam, chaptersData, QUERY_PARAMS_DATA, surahReduxValue),
+        isQueryParamDifferent,
+      };
```

Comment:
is there a more dynamic way to specify the `value` rather than adding this if condition check inside this dynamic function?